py/picca/io.py

```diff
@@ -1005,6 +1006,19 @@ def read_from_desi(in_dir, catalog, pk1d=None):
                     spec[key][w] = 0.
                 if f"{color}_RESOLUTION" in hdul:
                     spec["RESO"] = hdul[f"{color}_RESOLUTION"].read()
+                elif pk1d is not None:
+                    filename_truth=f"{in_dir}/{healpix//100}/{healpix}/truth-{in_nside}-{healpix}.fits"
+                    try:
+                        with fitsio.FITS(filename_truth) as hdul_truth:
+                            spec["RESO"] = hdul_truth[f"{color}_RESOLUTION"].read()
+                    except IOError:
+                        userprint(f"Error reading truth file {filename_truth}")   
+                    except KeyError:
+                        userprint(f"Error reading resolution from truth file for pix {healpix}")
+                    else:
+                        if not reso_from_truth:
```

Comment:
I don't see the usefulness of having this here. Maybe the whole block goes somewhere else? I mean the fact that you change it `reso_from_truth` to True does not imply anything here as the reading is done earlier. Maybe this block goes as an else at line 1009 and the elif there is really and if with and additional condiditon `if pk1d is not None and reso_from_truth:`

Reply:
this is only excecuted when it actually found a truth file with the right info (i.e. no exception on the command before), that's where the else comes in. The if is then in case the variable hasn't been set before to set it and print a message.

Reply:
This is explained by the previous block being properly indented